File: server/src/models/Checklist.ts
import mongoose, { Schema, Document } from 'mongoose';

export interface IChecklistItem {
  label: string;
  status: 'ok' | 'not_ok' | 'na';
  observation?: string;
}

export interface IChecklist extends Document {
  equipment: mongoose.Types.ObjectId;
  operator: mongoose.Types.ObjectId;
  date: Date;
  items: IChecklistItem[];
  isApproved: boolean;
  notes?: string;
  signatureUrl?: string;
  createdAt: Date;
  updatedAt: Date;
}

const ChecklistSchema: Schema = new Schema(
  {
    equipment: { type: Schema.Types.ObjectId, ref: 'Equipment', required: true },
    operator: { type: Schema.Types.ObjectId, ref: 'Operator', required: true },
    date: { type: Date, default: Date.now },
    items: [
      {
        label: { type: String, required: true },
        status: { type: String, enum: ['ok', 'not_ok', 'na'], required: true },
        observation: { type: String },
      },
    ],
    isApproved: { type: Boolean, required: true },
    notes: { type: String },
    signatureUrl: { type: String },
  },
  { timestamps: true }
);

export default mongoose.model<IChecklist>('Checklist', ChecklistSchema);

File: mobile/lib/models/checklist.dart
class Checklist {
  final String id;
  final String equipmentId;
  final String operatorId;
  final DateTime date;
  final List<ChecklistItem> items;
  final bool isApproved;
  final String? notes;
  final String? signatureUrl;

  Checklist({
    required this.id,
    required this.equipmentId,
    required this.operatorId,
    required this.date,
    required this.items,
    required this.isApproved,
    this.notes,
    this.signatureUrl,
  });

  factory Checklist.fromJson(Map<String, dynamic> json) {
    return Checklist(
      id: json['_id'] ?? '',
      equipmentId: json['equipment'] is Map ? json['equipment']['_id'] : json['equipment'],
      operatorId: json['operator'] is Map ? json['operator']['_id'] : json['operator'],
      date: DateTime.parse(json['date']),
      items: (json['items'] as List).map((i) => ChecklistItem.fromJson(i)).toList(),
      isApproved: json['isApproved'] ?? false,
      notes: json['notes'],
      signatureUrl: json['signatureUrl'],
    );
  }
}

class ChecklistItem {
  final String label;
  final String status; // 'ok', 'not_ok', 'na'
  final String? observation;

  ChecklistItem({
    required this.label,
    required this.status,
    this.observation,
  });

  factory ChecklistItem.fromJson(Map<String, dynamic> json) {
    return ChecklistItem(
      label: json['label'] ?? '',
      status: json['status'] ?? 'na',
      observation: json['observation'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'label': label,
      'status': status,
      'observation': observation,
    };
  }
}
